src/main/java/kg/jarkyn/cobspecserver/Server.java

```diff
@@ -28,6 +32,6 @@ private ClientSocket nextClient() {
     }
 
     private void respondTo(ClientSocket client) {
-        responseController.respond(client);
+        executor.execute(new Worker(client, responseController));
```

Comment:
I believe `executor.execute(new Worker(client, anotherResponseController));` will not make your tests fail. Not sure how valuable that test would be...